### PR TITLE
Cleanup of 'Merge-DatumArray'

### PR DESCRIPTION
**Pull request by raandree**

### CODE REVIEW

File: source/Private/Merge-DatumArray.ps1 [hunk 1]
```diff
@@ -1,58 +1,71 @@
 function Merge-DatumArray
 {
+    [OutputType([System.Collections.ArrayList])]
     [CmdletBinding()]
-    Param(
+    param (
+        [Parameter(Mandatory = $true)]
+        [hashtable[]]
```
Comment (gaelcolas) [inline source/Private/Merge-DatumArray.ps1:7]: too risky (loosing garantee of order)

File: source/Private/Merge-DatumArray.ps1 [hunk 2]
```diff
@@ -1,58 +1,71 @@
...
         $ReferenceArray,
 
+        [Parameter(Mandatory = $true)]
+        [hashtable[]]
```
Comment (gaelcolas) [inline source/Private/Merge-DatumArray.ps1:11]: too risky

File: source/Private/Merge-DatumArray.ps1 [hunk 3]
```diff
@@ -1,58 +1,71 @@
...
         $DifferenceArray,
 
-        $Strategy = @{ },
+        [Parameter()]
+        [hashtable]
```
Comment (gaelcolas) [inline source/Private/Merge-DatumArray.ps1:15]: ok

**Review by gaelcolas — CHANGES_REQUESTED** [review]

File: source/Private/Merge-DatumArray.ps1 (hunk 1 above)
Comment (raandree) [reply source/Private/Merge-DatumArray.ps1:7]: Which type would you choose? object[]?

File: source/Private/Merge-DatumArray.ps1 (hunk 3 above)
Comment (raandree) [reply source/Private/Merge-DatumArray.ps1:15]: Which type would you choose? object[]?

File: source/Private/Merge-DatumArray.ps1 (hunk 2 above)
Comment (raandree) [reply source/Private/Merge-DatumArray.ps1]: done

File: source/Private/Merge-DatumArray.ps1 (hunk 1 above)
Comment (gaelcolas) [reply source/Private/Merge-DatumArray.ps1]: Yep

**Review by gaelcolas — APPROVED** [review]

**Pull request merged**